Detectors/ITSMFT/ITS/reconstruction/include/ITSReconstruction/CookedTracker.h

```diff
@@ -91,64 +92,36 @@ class CookedTracker
   std::vector<CookedTrack> mSeeds; ///< Track seeds
 };
 
-// The helper classes
-class CookedTracker::ThreadData
-{
- public:
-  ThreadData();
-  ~ThreadData() {}
-  void resetSelectedClusters() { mI = 0; }
-  Int_t* Index() { return mIndex; }
-  Int_t& Nsel() { return mNsel; }
-  Int_t getNextClusterIndex()
-  {
-    while (mI < mNsel) {
-      Int_t ci = mIndex[mI++];
-      if (!mUsed[ci])
-        return ci;
-    }
-    return -1;
-  }
-  void useCluster(Int_t i) { mUsed[i] = kTRUE; }
- private:
-  ThreadData(const ThreadData&);
-  ThreadData& operator=(const ThreadData& tr);
-  Int_t mIndex[kMaxSelected];        ///< Indices of selected clusters
-  Int_t mNsel;                       ///< Number of selected clusters
-  Int_t mI;                          ///< Running index for the selected clusters
-  Bool_t mUsed[kMaxClusterPerLayer]; ///< Cluster usage flags
-};
-
 class CookedTracker::Layer
 {
  public:
   Layer();
+  Layer(const Layer&) = delete;
+  Layer& operator=(const Layer& tr) = delete;
 
-  void init(Int_t nThreads);
-  Double_t initInThread(Int_t first, Int_t last);
+  void init();
   Bool_t insertCluster(Cluster* c);
   void setR(Double_t r) { mR = r; }
   void unloadClusters();
-  void selectClusters(Int_t& i, Int_t idx[], Float_t phi, Float_t dy, Float_t z, Float_t dz);
+  void selectClusters(std::vector<Int_t> &s, Float_t phi, Float_t dy, Float_t z, Float_t dz);
   Int_t findClusterIndex(Double_t z) const;
   Float_t getR() const { return mR; }
   Cluster* getCluster(Int_t i) const { return mClusters[i]; }
   Float_t getXRef(Int_t i) const { return mXRef[i]; }
   Float_t getAlphaRef(Int_t i) const { return mAlphaRef[i]; }
   Float_t getClusterPhi(Int_t i) const { return mPhi[i]; }
-  Int_t getNumberOfClusters() const { return mN; }
+  Int_t getNumberOfClusters() const { return mClusters.size(); }
 
  protected:
-  Layer(const Layer&);
-  Layer& operator=(const Layer& tr);
+  enum {kNSectors=21};
 
   Float_t mR; ///< mean radius of this layer
 
-  Cluster* mClusters[kMaxClusterPerLayer]; ///< All clusters
-  Float_t mXRef[kMaxClusterPerLayer];      ///< x of the reference plane
-  Float_t mAlphaRef[kMaxClusterPerLayer];  ///< alpha of the reference plane
-  Float_t mPhi[kMaxClusterPerLayer];       ///< cluster phi
-  Int_t mN;                                ///< Total number of clusters
+  std::vector<Cluster*>mClusters;          ///< All clusters
+  std::vector<Float_t> mXRef;              ///< x of the reference plane
```

Comment:
Hi Sandro,

Yes, kMaxClusterPerLayer was a compile-time constant.  But, it was a very "generous" constant, set (with some quite extra margin) to withstand central Pb--Pb events.  For the cases of peripheral Pb--Pb, pPb and pp, it was just a waste memory...

Thank you for your comments !
Cheers,
Iouri.